ironfist/src/cpp/game/save.cpp

```diff
@@ -456,21 +456,22 @@ void game::LoadGame(char* filnam, int newGame, int a3) {
 			std::map<std::string, mapVariable> mapVariables;
 			for (ironfist_map::map_t::mapVariable_const_iterator it = mp->mapVariable().begin();
 				it != mp->mapVariable().end(); it++) {
-				mapVariable mapVar;
+				mapVariable *mapVar = new mapVariable;

```

Comment:
I have researched this issue.

If we were using C++11, we could use std::uniq_ptr, which would let us solve this problem.

After we get Sergei's build issues resolved, I'd like to take a look at officially adopting C++11 for the project (well, at least part of it: VS2015 only partially supports C++11). But.....that hasn't happened yet.

It seems you'll need to explicitly loop through the variables and delete them. And do similarly for the destructor  in scripting.h .